javaslang/src/main/java/javaslang/collection/Tree.java

```diff
@@ -334,11 +348,7 @@ default int leafCount() {
      * @return The number of nodes of this tree.
      */
     default int nodeCount() {
-        if (isEmpty()) {
-            return 0;
-        } else {
-            return 1 + getChildren().foldLeft(0, (count, child) -> count + child.nodeCount());
-        }
+        return length();
```

Comment:
Hmm, do you have an article which explains this in more detail, sounds interesting. If not, you should write one (maybe even in the Javaslang documentation, if you think it adds functional value to Java) :)